core/src/main/java/org/lflang/generator/c/TypeParameterizedReactor.java

```diff
@@ -94,18 +145,13 @@ public InferredType resolveType(InferredType t) {
    * Return a name that is unique to this TypeParameterizedReactor (up to structural equality) and
    * that is prefixed with exactly one underscore and that does not contain any upper-case letters.
    */
-  public synchronized String uniqueName() {
-    String name = reactor.getName().toLowerCase();
-    if (uniqueNames.containsKey(this)) return uniqueNames.get(this);
-    if (nameCounts.containsKey(name)) {
-      int currentCount = nameCounts.get(name);
-      nameCounts.put(name, currentCount + 1);
-      uniqueNames.put(this, "_" + name + currentCount);
-      return uniqueName();
-    }
-    nameCounts.put(name, 1);
-    uniqueNames.put(this, "_" + name);
-    return uniqueName();
+  public String uniqueName() {
+    var resolved = ASTUtils.toDefinition(reactor);
+    return "_"
+        + uniqueName(resolved)
+        + typeParams.stream()
+            .map(it -> typeArgs.get(it).getId()) // FIXME: may be more than just an ID
```

Comment:
Mysterious comment. What does it mean?

Reply:
It means that a type can consist of multiple words or can be escaped in `{= =}`. There is already an open issue for this: it is the first item listed in https://github.com/lf-lang/lingua-franca/issues/1765.